pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -665,6 +665,7 @@ func (configgen *ConfigGeneratorImpl) createGatewayHTTPFilterChainOpts(node *mod
 	// We know that this is a HTTPS server because this function is called only for ports of type HTTP/HTTPS
 	// where HTTPS server's TLS mode is not passthrough and not nil
 	http3Enabled := transportProtocol == istionetworking.TransportProtocolQUIC
+	mtlsGatewayMxFilter := server.GetTls().GetMode() == networking.ServerTLSSettings_ISTIO_MUTUAL
```

Comment:
We need many levels of protection - can't blinding add it to all mtls listeners.